### PR TITLE
New email

### PR DESCRIPTION
**Pull request by emshee13**
Please review carefully, I didn't have some of the newer changes to master.  Please test confirmation and password reset, but to do so you'll need to go into pipeline_ui/proxy/conf and change proxy_set_header value (line 6) to your own IPv4 (use ipconfig command to find your IP address)

### CODE REVIEW

**Comment by sarfrazali1947** [comment]
I've tested registration email and password reset email features. They are working as expected

**Comment by AJTDaedalus** [comment]
Merging changes in master into this was a bit of a nightmare.  Can a couple of people please check all functionality on both development and production (docker) versions of the server?

File: proxy/conf [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-    listen       80;
-    server_name  app;
-    location / {
-        proxy_pass   http://app:8000;
+    listen               80;
+	listen               587 ssl;
+    server_name          app;
+    location / {  
+        proxy_pass                http://app:8000;
+        proxy_set_header          Host 172.20.0.1;
```
Comment (LSMachuca) [inline proxy/conf]:
```suggestion
        proxy_set_header          Host $http_host;
```
With the above change you don't need to look for your IP address. It should work in all cases

File: code/app/models.py [hunk 2]
```diff
@@ -78,7 +78,46 @@ def password(self, password):
...
+    def generate_confirmation_token(self, expiration=30800):
+        """Generate a confirmation token to email a new user."""
+        s = Serializer(current_app.config['SECRET_KEY'], salt='email-confirm')
+        return s.dumps(self.id)
+
+    def confirm_account(self, token):
+        """Verify that the provided token is for this user's id."""
+        s = Serializer(current_app.config['SECRET_KEY'])
+        try:
+            email = s.loads(token, salt='email-confirm', max_age=60)
```
Comment (LSMachuca) [inline code/app/models.py]:
```suggestion
            email = s.loads(token, salt='email-confirm', max_age=600)
```
previous setting of 60 sec was not enough time to receive email and click to confirm. Changed to 10 min, but prob should be longer for real users.

**Review by LSMachuca — COMMENTED** [review]
Finished testing, everything worked. Please see suggestions.

**Pull request merged**